src/MessagePack/Formatters/EnumAsStringFormatter`1.cs

```diff
@@ -3,14 +3,15 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 using System.Reflection;
 using System.Runtime.Serialization;
 using MessagePack.Internal;
 
 namespace MessagePack.Formatters
 {
     // Note:This implementation is 'not' fastest, should more improve.
-    public sealed class EnumAsStringFormatter<T> : IMessagePackFormatter<T>
+    public sealed class EnumAsStringFormatter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields)] T> : IMessagePackFormatter<T>
```

Comment:
Note: in net9 we changed our trimming rules such that when `T` is guaranteed to be an Enum (like it is here), it no longer warns when you try to get the fields of the enum type. I don't think adding this attribute is causing you problems now, but just an FYI. See https://github.com/dotnet/runtime/pull/100814 for more info.